tokio/src/runtime/context.rs

```diff
@@ -47,6 +47,19 @@ cfg_rt_core! {
     }
 }
 
+cfg_syscall! {
+    use std::sync::Arc;
+    use crate::syscall::Syscalls;
+
+    #[allow(dead_code)] // Unused for now, gated by the unstable flag
+    pub(crate) fn syscalls() -> Option<Arc<dyn Syscalls>> {
```

Comment:
It seems like there is currently a lot of code that just needs a reference to the `Syscalls`, rather than an owned `Arc<dyn Syscalls>`...WDYT about also adding a function for accessing it by-ref, without an arc bump?

Something like:
```rust
    pub(crate) fn with_syscalls<T>(f: impl FnOnce(&dyn Syscalls) -> T) -> Option<T> {
        CONTEXT.with(|ctx| { 
            ctx.borrow().map(|ctx| f(&ctx.syscalls))
        })
    }
```
in addition to `syscalls`.